Documentation/network/servicemesh/ingress.rst

```diff
@@ -87,6 +87,13 @@ Additionally, cloud-provider specific annotations for the LoadBalancer service
 are supported. Please refer to the `Kubernetes documentation <https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer>`_
 for more details.
 
+.. warning::
+
+    The  ``Ingress`` and ``Gateway API`` features in ``Cilium`` require the layer 7 proxy to be enabled in order to function properly. 
+    If the layer 7 proxy is not enabled, these features will not work as intended. Before using the Ingress and Gateway API, 
+    please ensure that your Cilium installation has the layer 7 proxy enabled and properly configured. If you have any questions or concerns, 
```

Comment:
As discussed on Slack, it would be great to tell readers how to have this enabled (and given that the relevant option is true by default, Joe suggested to turn the warning into a mere info block.

Reply:
Very helpful. I will improve this.